src/runtime/auto-prompt-manager.js

```diff
@@ -345,10 +430,14 @@ export class AutoPromptManager {
   /**
    * Determines whether a larger, blocking prompt should be shown.
    * @param {!../api/entitlements.Entitlements} entitlements
+   * @param {!boolean} hasPotentialAudienceAction
    * @returns {boolean}
    */
-  shouldShowLockedContentPrompt_(entitlements) {
-    return this.pageConfig_.isLocked() && !entitlements.enablesThis();
+  shouldShowBlockingPrompt_(entitlements, hasPotentialAudienceAction) {
+    return (
+      (this.pageConfig_.isLocked() || hasPotentialAudienceAction) &&
```

Comment:
Why is this needed?

Reply:
So the `isLocked` is used for premium content (i.e. not `isAccessibleForFree`). Contribution pages are all marked as `isAccessibleForFree: true` but we still want to show the action if we have one available so I pushed it into here.

This method also checks to see if the user is entitled to the page, in which we never want to show a blocking prompt

Reply:
I want to confirm if I understand correctly why we need to add the `hasPotentialAudienceAction` check here:

- For subscription model, this check should not change anything.
- For contribution model, this is added so that we don't cap regwall / newsletter? For example, if the user dismissed the contribution prompt, `shouldShowAutoPrompt` would be false. So we will need to rely on `shouldShowBlockingPrompt_()` to show the audience action and therefore we need to add the `hasPotentialAudienceAction` check here. A potential downside is that this action will not be capped by any frequency setting and user will keep seeing the action prompt until they explicitly dismiss it (which I guess it's fine since the prompt is blocking the user to read the content and if the user does want to read the content they will either dismiss or complete the prompt).

Reply:
Both are correct. If we can show a contribution prompt, we should; regardless of actions hence the additional check in this flow. You are also correct that the user will keep seeing a regwall/newsletter unless it is dismissed explicitly. I imagine this could be useful in the case of someone bouncing on the page and never seeing the prompt again